apps/dot_ext/oauth2_validators.py

```diff
@@ -109,13 +108,8 @@ def __call__(self, value):
             raise ValidationError('Redirect URIs must not contain fragments')
         scheme, netloc, path, query, fragment = urlsplit(value)
 
-        # Fix the mobile endpoint validation
-        # Allow 2 character alpha plus 8 numerics
-        if scheme.lower() in self.allowed_schemes:
-            pass
-
-        elif scheme.lower() not in self.allowed_schemes:
-            raise ValidationError('Invalid Redirect URI:[%s]' % scheme.lower())
+        if scheme.lower() not in self.allowed_schemes:
+            raise ValidationError('Invalid Redirect URI scheme: %s' % scheme.lower())
 
 
 def validate_uris(value):
```

Comment:
Can you add some tests around this? Asserting something like `https, http` work and something else raises the extected `Exception`, and a url with a `#` raises an `Exception`.